<?xml version="1.0"?>
<launch>

  <include file="$(find audibot_final_project)/launch/two_audibot_road_world.launch" />

  <include file="$(find audibot_lane_detection)/launch/lane_detection.launch" >
    <arg name="nodelet" value="true" />
    <arg name="robot_name" value="a1" />
  </include>

  <include file="$(find audibot_lane_detection)/launch/lane_detection.launch" >
    <arg name="nodelet" value="true" />
    <arg name="robot_name" value="a2" />
  </include>

  <group ns="a1" >
    <node pkg="audibot_path_following" type="audibot_path_following" name="path_following" output="screen" >
      <param name="speed" value="23.0" />
      <remap from="/a1/cmd_vel" to="/a1/path/cmd_vel"/>
    </node>
    <node pkg="audibot_twist_controller" type="audibot_twist_node" name="twist_control" output="screen" />
  </group>

  <group ns="a2" >
    <node pkg="audibot_path_following" type="audibot_path_following" name="path_following" output="screen" >
      <param name="speed" value="20.0" />
      <remap from="/a2/cmd_vel" to="/a2/path/cmd_vel"/>
    </node>
    <node pkg="audibot_twist_controller" type="audibot_twist_node" name="twist_control" output="screen" />
  </group>

  <!-- Launch our project node here!!! -->
  <arg name="level"/>

  <group if="$(eval level == 'level1')" >
    <node pkg="audibot_final_project" type="audibot_final_project" name="audibot_final_project" output="screen" >
      <param name="level" type="string" value="level1" />
    </node>
  </group>

  <group if="$(eval level == 'level2')" >
    <node pkg="audibot_final_project" type="audibot_final_project" name="audibot_final_project" output="screen" >
      <param name="level" type="string" value="level2" />
    </node>
  </group>

  <group if="$(eval level == 'level3')" >
    <node pkg="audibot_final_project" type="audibot_final_project" name="audibot_final_project" output="screen" >
      <param name="level" type="string" value="level3" />
    </node>
  </group>

  <group if="$(eval level == 'pcl')" >
    <node pkg="audibot_final_project" type="audibot_final_project" name="audibot_final_project" output="screen" >
      <param name="level" type="string" value="pcl" />
    </node>
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find audibot_final_project)/rviz/simulation.rviz" />
    <param name="robot_description" textfile="$(find audibot_final_project)/urdf/audibot_description.urdf" />
  </group>

</launch>